src/bank02.asm

```diff
@@ -8130,29 +8126,197 @@ titleScreenLicenseText:
-    TXT  "  the way to the<00>"                        ;; 02:7f93 ...........??????
-    TXT  " Tree of Mana to<00>"                        ;; 02:7fa4 ?????????????????
-    TXT  "  get the mighty<00>"                        ;; 02:7fb5 ?????????????????
-    TXT  " power to conquer<00>"                       ;; 02:7fc6 ??????????????????
-    TXT  "    the world.<00>"                          ;; 02:7fd8 ???????????????
-    TXT  "<00>"                                        ;; 02:7fe7 ?
-    db   $00, $00, $00, $00, $00, $00, $00, $00        ;; 02:7fe8 ????????
-    db   $00, $00, $00, $00, $01, $ff, $ff, $ff        ;; 02:7ff0 ????????
-    db   $ff, $ff, $ff, $ff, $ff, $ff, $ff, $ff        ;; 02:7ff8 ????????
+    TXT  "<00>"                                        ;; 02:7fd8 ?
+    db   $01                                           ;; 02:7fd9 ?
+
+; Draws left aligned number HL as modified tiles at WRAM position DE
+; C holds the drawing mode in the first nibble (0 for shift, 1 for shift/swap)
+; C holds number of max digits in the second nibble
+drawLeftAlignedNumberInWRAM:
+    ld   B, $00 ; digit counter
+
+    ; Housekeeping, get a copy of proper WRAM starting
+    ;  location on the stack for later use
+    push BC
+    push DE
+
+    ; Load up the digits on the stack
+.divmod_and_store_number:
+    ld   A, $0a
+    push BC
+    push DE
+    call divMod
+    pop  DE
+    pop  BC
+    push AF
+    inc  B
+    ld   A, H
+    or   A, L
+    jr   NZ, .divmod_and_store_number
+
+    ; Reduce second nibble of C by num digits found
+    ; Remainder provides the number of tiles to clear
+    ; after we write the digits
+    ld   A, C
+    sub  A, B
+    ld   C, A
+
+    ; Increment tile buffer pointer (DE) if in swap mode
+    ; since we will be swapping back 4px to the tile before
+    ld   H, D
+    ld   L, E
+    ld   D, $00
+    and  A, $10
+    ld   E, A
+    add  HL, DE
+    ld   D, H
+    ld   E, L
+
+    ; Stack holds B digits to pop, transfer them
+ .grab_digit_and_transfer:
+    ld   HL, gfxStatusBar+$100 ; start of number tiles
+    pop  AF ; grab digit
+    swap A
+    push DE
+    ld   D, $00
+    ld   E, A
+    add  HL, DE
+    pop  DE
+    ld   A, BANK(gfxStatusBar)
+    push BC
+    ld   B, $10
+    call copyAndRotateBankAfromHLtoDE
+    pop  BC
+    dec  B
+    jr   NZ, .grab_digit_and_transfer
+
+    ; Clear remaining tiles
+    ld   A, C
+    and  A, $0f
+    jr   Z, .check_swap
+
+    ld   H, D
+    ld   L, E
+    ld   B, A
+    xor  A, A
+ .clear_remaining_tiles:
+    call memsetTileWithA
+    dec  B
+    jr   NZ, .clear_remaining_tiles
+
+    ; Shifted tiles are in WRAM, check if swap needed
+ .check_swap:
+    pop  DE
+    ld   L, C
+    pop  BC
+    bit  4, C
+    ret  Z
+
+    ; We need to swap back tiles, prepare...
+    push BC
+    push DE
+    ld   A, C
+    sub  A, L
+    ld   B, A ; B holds number of tiles to swap
+    call swapBackHalfTiles
```

Comment:
Is this a candidate for inlining? Do you see it being used other places in the future?

Reply:
Absolutely it is, I just didn't like the size of the function before.

Reply:
I'm personally more okay with large functions than with functions that are called just once. Lets keep this separate for now until I know whether I'm going to scatter these across bank 2, or put them into a new bank.